main/src/org/jetbrains/bsp/bazel/resolvers/QueryResolver.java

```diff
@@ -0,0 +1,28 @@
+package org.jetbrains.bsp.bazel.resolvers;
+
+import com.google.devtools.build.lib.query2.proto.proto2api.Build;
+import java.io.IOException;
+import java.util.concurrent.Semaphore;
+
+public class QueryResolver {
+
+  private final ProcessResolver processResolver;
+  private final Semaphore processLock;
```

Comment:
so now we have more than one semaphore... are you sure that it works the same way as before?